cobalt/devinfra/kokoro/bin/dind_build.sh

```diff
@@ -72,12 +72,21 @@ pipeline () {
   ##############################################################################
   cd "${gclient_root}/src"
   cobalt/build/gn.py -p "${TARGET_PLATFORM}" -C "${CONFIG}" \
-    --script-executable=/usr/bin/python3
-  if [[ "${TARGET_PLATFORM}" =~ "android" ]]; then
-    echo "is_cobalt_on_google3 = true" >> out/${TARGET_PLATFORM}_${CONFIG}/args.gn
-  fi
+    --script-executable=/usr/bin/python3 --args="${EXTRA_GN_ARGUMENTS:-}"
   autoninja -C "out/${TARGET_PLATFORM}_${CONFIG}" ${GN_TARGET}  # GN_TARGET may expand to multiple args
 
+  # Build targets used for testing.
+  if [[ "${TARGET_PLATFORM}" =~ "android-arm" && "${CONFIG}" == "qa" ]]; then
+    # Build Cobalt.apk without `${EXTRA_GN_ARGUMENTS}` for e2e testing.
+    cobalt/build/gn.py -p "${TARGET_PLATFORM}" -C "${CONFIG}" \
+      --script-executable=/usr/bin/python3
+    autoninja -C "out/${TARGET_PLATFORM}_${CONFIG}_tmp" "android:cobalt_apk"
+
+    # Replace the APK from the original build.
```

Comment:
@zhongqiliang Can we replace Cobalt.apk built with `is_cobalt_on_google3=true` here with one built without the flag?

Reply:
Discussed offline, the apk is not used for anything.